File: rb/features.rb
# Maplestory SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MaplestoryFeatures
  def self.make_feature(name)
    case name
    when "base"
      MaplestoryBaseFeature.new
    when "test"
      MaplestoryTestFeature.new
    else
      MaplestoryBaseFeature.new
    end
  end
end

File: py/features.py
# Maplestory SDK feature factory

from feature.base_feature import MaplestoryBaseFeature
from feature.test_feature import MaplestoryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MaplestoryBaseFeature(),
        "test": lambda: MaplestoryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
